aten/src/ATen/native/sparse/cuda/SparseCUDABlas.cu

```diff
@@ -71,6 +80,119 @@ cusparseOperation_t convertTransToCusparseOperation(char trans) {
   }
 }
 
+#if !defined(MSC_VER) && defined(__CUDACC__) && CUSPARSE_VERSION >= 10301 // CUDA release >= 10.2 and not windows
+
+template<typename T> 
+void csrmm2(
```

Comment:
Is this code copy pasted from somewhere?

Reply:
If you mean the macro, I wrote the code and tested it.

- CUDA 10.2 guard is because the new function signature gives inaccurate results in 10.1 
- `__CUDACC__` guard is to prevent build failures on ROCm
- Windows guard is because the new signature may be not supported in current CUDA version (see the PR comment) 

The first few lines of this function (cusparseOperation_t and TORCH_CHECK) are copied from the old version of this function.

Reply:
I'm referring to the full contents of the `csrmm2` template function. (Basically, I need to know if I should be reviewing this as if it were new code, or if I am just checking it is the same as something we had before.)

Reply:
Ohh, thanks! These are all new code I wrote.